fix(curriculum): update db method used in implementation-of-social-authentication-iii

kostaslamo opened the pull request:
Updating the method used at implementation-of-social-authentication-iii, both for english and chinese

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [ ] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #39592

<!-- Feel free to add any additional description of changes below this line -->

Updated to use the method `findOneAndUpdate()` instead of the deprecated `findAndModify()`.
Also my prettier has prettified the two files, used .prettierrc in the root projects folder as configuration file.

Thank you.

kostaslamo committed:
fix: update method used from findAndModify to findOneAndUpdate

Updating the method used at implementation-of-social-authentication-iii, both for english and chinese

ShaunSHamilton reviewed (changes requested):
Thank you, for being eager to contribute. However, i am not sure it is a good idea to update both languages.

Specifically, because the paragraph underneath the changed code **needs to be updated as well** to reflect the change from `findAndModify`

@RandellDawson Is it alright for the Chinese changes to also be included in this PR?

kostaslamo committed:
fix: changed text findAndModify to findOneAndUpdate

kostaslamo commented:
I have also added a commit changing the texts from findAndModify to findOneAndUpdate in english and chinese. Hope now could be merged.

RandellDawson commented:
@kostaslamo The checkbox labelled `All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.` is meant to be checked off, so you will need to create a separate PR for the Chinese changes.

Please remove the Chinese file from this PR.

Thank you.

kostaslamo committed:
fix: removed chinese file

kostaslamo commented:
> @kostaslamo The checkbox labelled `All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.` is meant to be checked off, so you will need to create a separate PR for the Chinese changes.
> 
> Please remove the Chinese file from this PR.
> 
> Thank you.

Ok, i' ve got it. Removed the Chinese file from the PR.

Thank you.

RandellDawson commented on curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication-iii.english.md:
````diff
@@ -6,34 +6,40 @@ forumTopicId: 301558
 ```
 
-With a findAndModify, it allows you to search for an object and update it, as well as insert the object if it doesn't exist and receive the new object back each time in our callback function. In this example, we always set the last_login as now, we always increment the login_count by 1, and only when we insert a new object(new user) do we populate the majority of the fields. Something to notice a ...
+With a findOneAndUpdate, it allows you to search for an object and update it, as well as insert the object if it doesn't exist and receive the new object back each time in our callback function. In this example, we always set the last_login as now, we always increment the login_count by 1, and only when we insert a new object(new user) do we populate the majority of the fields. Something to notic ...
````
While we are changing this one, let's clean up the wording:
```suggestion
`findOneAndUpdate` allows you to search for an object and update it. If the object doesn't exist, it will be inserted and made available to the callback function. In this example, we always set `last_login`, increment the `login_count` by `1`, and only populate the majority of the fields when a new object (new user) is inserted. Notice the use of default values. Sometimes a profile returned won't have all the information filled out or the user will keep it private.  In this case, you handle it to prevent an error.
```

kostaslamo replied:
Great, makes more sense. Commited.

RandellDawson reviewed (changes requested):
See my one suggestion.

kostaslamo committed:
Update curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication-iii.english.md

Co-authored-by: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

RandellDawson commented:
LGTM now.  @Sky020 What do you think?

ShaunSHamilton reviewed (approved):
This LGTM :thumbsup:

Thanks, for the wording changes, Randell.

Sky020 commented:
Thank you, for your contribution @kostaslamo  

Also, congrats on your first PR! We look forward to future contributions.

kostaslamo commented:
> Thank you, for your contribution @kostaslamo
> 
> Also, congrats on your first PR! We look forward to future contributions.

Thank you very much @RandellDawson @Sky020 . Looking also forward for future contributions. Maybe pm someone, to discuss for a task i could get assigned?

RandellDawson commented:
@kostaslamo Just look at the list of GitHub issues and take on it.
